lib/websocket.js

```diff
@@ -225,7 +225,13 @@ class WebSocket extends EventEmitter {
     }
 
     if (this.readyState === WebSocket.CLOSING) {
-      if (this._closeFrameSent && this._closeFrameReceived) this._socket.end();
+      if (
+        this._closeFrameSent &&
+        (this._closeFrameReceived || this._receiver._writableState.errorEmitted)
```

Comment:
@pimterry I think that adding `this._receiver._writableState.finished` here and below along with the `-2` ready state would also address https://github.com/websockets/ws/pull/1902#issuecomment-867889227.

Anyway I would prefer to keep it as is because:

1. A socket `'end'` event with no close frame means that the remote peer is doing something wrong.
1. We are consistent with browsers behavior. The connection is closed immediately with no close frame sent back.